CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,359 @@
+                        }
+                        """)
+
+                  } else {
+                    DeclSyntax("""
+                        public init(_ node: \(raw: choice.typeName)) {
+                          self = .\(raw: choice.swiftName)(node)
+                        }
+                        """)
+
+                  }
+                }
+
+                try! InitializerDeclSyntax("public init?<S: SyntaxProtocol>(_ node: S)") {
+                  for choice in child.nodeChoices {
+                    StmtSyntax("""
+                    if let node = node.as(\(raw: choice.typeName).self) {
+                      self = .\(raw: choice.swiftName)(node)
+                      return
+                    }
+                    """)
+                  }
+                  StmtSyntax("return nil")
+                }
+
+                try! VariableDeclSyntax("public static var structure: SyntaxNodeStructure") {
+                  ReturnStmtSyntax(
+                    expression: FunctionCallExprSyntax(callee: ExprSyntax(".choices")) {
+                      TupleExprElementSyntax(
+                        expression: ArrayExprSyntax {
+                          for choice in child.nodeChoices {
+                            ArrayElementSyntax(
+                              expression: ExprSyntax(".node(\(raw: choice.typeName).self)"))
+                          }
+                        }
+                      )
+                    }
+                  )
+                }
+              }
+            }
+          }
+
+          // ==============
+          // Initialization
+          // ==============
+
+          DeclSyntax("public let _syntaxNode: Syntax")
+
+          DeclSyntax("""
+            public init?<S: SyntaxProtocol>(_ node: S) {
+              guard node.raw.kind == .\(raw: node.swiftSyntaxKind) else { return nil }
+              self._syntaxNode = node._syntaxNode
+            }
+            """)
+
+          DeclSyntax("""
+            /// Creates a `\(raw: node.name)` node from the given `SyntaxData`. This assumes
+            /// that the `SyntaxData` is of the correct kind. If it is not, the behaviour
+            /// is undefined.
+            internal init(_ data: SyntaxData) {
+              assert(data.raw.kind == .\(raw: node.swiftSyntaxKind))
+              self._syntaxNode = Syntax(data)
+            }
+            """)
+
+          try! InitializerDeclSyntax("\(raw: node.generateInitializerDecl(optionalBaseAsMissing: false, currentIndentation: "  "))") {
+            let parameters = node.children.map { $0.swiftName }.joined(separator: ", ")
+
+            VariableDeclSyntax(
+              leadingTrivia: [
+                .docLineComment("// Extend the lifetime of all parameters so their arenas don't get destroyed"),
+                .newlines(1),
+                .docLineComment("// before they can be added as children of the new arena."),
+                .newlines(1)
+              ],
```

Comment:
Trivia is now expressible by string literals, so this can just be

```swift
leadingTrivia: """
// Extend the lifetime of all parameters so their arenas don't get destroyed
// before they can be added as children of the new arena.
"""
```